File: app/src/main/java/com/github/pockethub/android/ui/item/TextItem.java
package com.github.pockethub.android.ui.item;

import android.support.annotation.IdRes;
import android.support.annotation.LayoutRes;
import android.support.annotation.NonNull;
import android.widget.TextView;

import com.xwray.groupie.Item;
import com.xwray.groupie.ViewHolder;

public class TextItem extends Item<ViewHolder> {

    private final int layoutId;
    private final int textViewId;
    private final CharSequence text;

    public TextItem(@LayoutRes int layoutId, @IdRes int textViewId, CharSequence text) {
        super(text.hashCode());
        this.layoutId = layoutId;
        this.textViewId = textViewId;
        this.text = text;
    }

    @Override
    public void bind(@NonNull ViewHolder viewHolder, int position) {
        TextView textView = viewHolder.getRoot().findViewById(textViewId);
        textView.setText(text);
    }

    @Override
    public int getLayout() {
        return layoutId;
    }
}

File: app/src/main/java/com/github/pockethub/android/ui/item/TextItem.kt
package com.github.pockethub.android.ui.item

import androidx.annotation.IdRes
import androidx.annotation.LayoutRes
import android.widget.TextView
import com.xwray.groupie.kotlinandroidextensions.Item
import com.xwray.groupie.kotlinandroidextensions.ViewHolder

class TextItem(
        @param:LayoutRes private val layoutId: Int,
        @param:IdRes private val textViewId: Int,
        private val text: CharSequence
) : Item(text.hashCode().toLong()) {

    override fun getLayout() = layoutId

    override fun bind(holder: ViewHolder, position: Int) {
        val textView = holder.root.findViewById<TextView>(textViewId)
        textView.text = text
    }
}
